l5kit/l5kit/dataset/select_agents.py

```diff
@@ -112,82 +99,64 @@ def get_valid_agents(
 
             # ==== POINT-WISE FILTERS
             if not of_interest[global_agent_idx]:
-                frame_lost = get_missing_frame_num(agents_dict[agent["track_id"]], agents_selected_mask)
-                report["reject_th_agent_filter_probability_threshold"] += frame_lost
+                update_mask(agents_mask, agents_dict[agent["track_id"]][:-1])
+                report["reject_th_agent_filter_probability_threshold"] += 1
                 agents_dict[agent["track_id"]] = []
                 continue
+
             if not in_av_distance(frame["ego_translation"], agent["centroid"], th_distance_av):
-                frame_lost = get_missing_frame_num(agents_dict[agent["track_id"]], agents_selected_mask)
-                report["reject_th_agent_distance_av_threshold"] += frame_lost
+                update_mask(agents_mask, agents_dict[agent["track_id"]][:-1])
+                report["reject_th_AV_distance"] += 1
                 agents_dict[agent["track_id"]] = []
                 continue
 
             # ==== COUPLE-WISE FILTERS
             if len(agents_dict[agent["track_id"]]) > 1:
                 p_frame_idx, p_global_agent_idx, p_agent = agents_dict[agent["track_id"]][-2]  # get prev element
-                frame_lost = get_missing_frame_num(agents_dict[agent["track_id"]][:-1], agents_selected_mask)
 
                 if not in_consecutive_frame(frame_idx, p_frame_idx):
-                    report["reject_th_hole"] += frame_lost
+                    update_mask(agents_mask, agents_dict[agent["track_id"]][:-1])
```

Comment:
Could think about shortening this, as there are nearly 3 duplicate lines for each case.

Reply:
because we need to fill the report we won't save any lines sadly :(